File: Java/SelectionSort.java
import java.util.*;
class selectionSort
{
    public static void main(String args[])
    {
        int arr[] = {2,3,1,5,4};
        
        for(int i = 0; i < arr.length-1; i++)
        {
            int d = i;
            for(int j = i+1; j < arr.length; j++)
            {
                if(arr[j] < arr[d])
                {
                    d = j;
                }
            }
            int t = arr[i];
            arr[i] = arr[d];
            arr[d] = t;
        }
        for(int i = 0; i < arr.length; i++)
        {
            System.out.print(arr[i]+" ");
        }
    }
}

File: Python/SelectionSort.py
def selection_sort(arr):
    # Get the length of the array
    n = len(arr)
    
    # Loop through the array
    for i in range(n):
        # Assume the current index is the minimum
        min_idx = i
        
        # Loop through the unsorted part of the array
        for j in range(i+1, n):
            # If a smaller element is found, update the minimum index
            if arr[j] < arr[min_idx]:
                min_idx = j
        
        # Swap the minimum element with the first element of the unsorted part
        arr[i], arr[min_idx] = arr[min_idx], arr[i]
    
    # Return the sorted array
    return arr